src/test/java/org/junitpioneer/jupiter/ReportEntryExtensionTest.java

```diff
@@ -69,20 +82,158 @@ void emptyValue_fails() {
 	}
 
 	@Test
+	@DisplayName("logs each value as individual entry when annotation is repeated")
 	void repeatedAnnotation_logEachKeyValuePairAsIndividualEntry() {
 		ExecutionEventRecorder recorder = executeTestsForMethod(ReportEntriesTest.class, "repeatedAnnotation");
 
 		List<Map<String, String>> reportEntries = reportEntries(recorder);
 
 		assertAll("Verifying report entries " + reportEntries, //
 			() -> assertThat(reportEntries).hasSize(3),
-			() -> assertThat(reportEntries).extracting(entry -> entry.size()).containsExactlyInAnyOrder(1, 1, 1),
+			() -> assertThat(reportEntries).extracting(Map::size).containsExactlyInAnyOrder(1, 1, 1),
 			() -> assertThat(reportEntries)
 					.extracting(entry -> entry.get("value"))
 					.containsExactlyInAnyOrder("suddenly there came a tapping", "As if some one gently rapping",
 						"rapping at my chamber door"));
 	}
 
+	@Nested
+	@DisplayName("with explicitly set when parameter")
```

Comment:
maybe put the `when` into quotes or so. As of now it reads a bit rough (at least to me)